docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
+> which can be built into either initrd or rootfs guest images.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
-has a highly optimized boot path.
+The default packaged root filesystem image ("rootfs"), sometimes
+referred to as the "mini O/S", is a highly optimized container
+bootstrap system based on [Clear Linux](https://clearlinux.org). It
+provides an extremely minimal environment and has a highly optimized
+boot path.
+
+The only services running in the context of the mini O/S are:
+
+| Process | Environment | systemd service? | Notes |
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
+
+See also the [process overview](#process-overview).
 
-The only services running in the context of the mini O/S are the init daemon
-(`systemd`) and the [Agent](#agent). The real workload the user wishes to run
-is created using libcontainer, creating a container in the same manner that is done
-by `runc`.
+> **Note:** The container workload is running inside a full container
+> environment which itself is running within a VM environment.
 
-For example, when `ctr run -ti ubuntu date` is run:
+For example, when the user runs `ctr run -ti ubuntu date`:
 
 - The hypervisor will boot the mini-OS image using the guest kernel.
 - `systemd`, running inside the mini-OS context, will launch the `kata-agent` in
   the same context.
 - The agent will create a new confined context to run the specified command in
   (`date` in this example).
 - The agent will then execute the command (`date` in this example) inside this
-  new context, first setting the root filesystem to the expected Ubuntu\* root
+  new context, first setting the root filesystem to the expected Ubuntu root
   filesystem.
 
 #### Initrd image
 
-A compressed `cpio(1)` archive, created from a rootfs which is loaded into memory and used as part of the Linux startup process. During startup, the kernel unpacks it into a special instance of a `tmpfs` that becomes the initial root filesystem.
+The initrd image is a compressed `cpio(1)` archive, created from a
+rootfs which is loaded into memory and used as part of the Linux
+startup process. During startup, the kernel unpacks it into a special
+instance of a `tmpfs` mount that becomes the initial root filesystem.
+
+The only service running in the context of the initrd is the
+[Agent](#agent) as the init daemon. The real workload the user wishes
+to run is created using libcontainer, creating a container in the same
+manner that is done by `runc`.
+
+| Process | Environment | Notes |
```

Comment:
I don't really understand what environment means here.  The terms VM root and VM container don't make sense to me.  I think an extra column "User Accessible" or some other name to indicate what the user can interact with and what they can't would help.

Reply:
This is hard to not confuse.   Would it make sense to have a glossary of terms at the beginning of the document or this section to state what it is?  I'll have to think about this one a bit more.  You could expand the table out to include the binaries that run on the host as well but im not sure all the columns make sense and isn't necessarily relevant to this section. 

One thing that still confuses me is the `sh(1)` as the container workload.  When I think of workload I think of envoy or nginx.